src/js/device/DeviceList.js

```diff
@@ -345,8 +210,177 @@ export const getDeviceList = async (): Promise<DeviceList> => {
     try {
         await list.init();
         return list;
-    } catch(error) {
-        console.error("INITERROR", error);
+    } catch (error) {
+        console.error('INITERROR', error);
         throw ERROR.NO_TRANSPORT;
     }
+};
+
+// Helper class for creating new device
+class CreateDeviceHandler {
+
+    descriptor: DeviceDescriptor;
+    list: DeviceList;
+    path: string;
+
+    constructor(descriptor: DeviceDescriptor, list: DeviceList): Promise<void> {
+        logger.debug('Creating Device', descriptor);
+        this.descriptor = descriptor;
+        this.list = list;
+        this.path = descriptor.path.toString();
+    }
+
+    // main logic
+    async handle() {
+        // creatingDevicesDescriptors is needed, so that if *during* creating of Device,
+        // other application acquires the device and changes the descriptor,
+        // the new unacquired device has correct descriptor
+        this.list.creatingDevicesDescriptors[this.path] = this.descriptor;
+
+        try {
+            // "regular" device creation
+            await this._takeAndCreateDevice();
+        } catch (error) {
+            logger.debug('Cannot create device', error);
+
+            // if (error.message === ERROR.WRONG_PREVIOUS_SESSION_ERROR_MESSAGE) {
```

Comment:
This did not actually ever happen.

Before this refactor, there was this code there

```
               let existed: ?Device = this.devices[path];
               if (existed instanceof Device && !existed.isUnacquired()) {
                   existed.disconnect(); // TODO: is it necessary?
                   delete this.devices[path];
                   existed = null;
               }
               if(existed === null) {
                   device = await this._createUnacquiredDevice(transport, descriptor);
                   this.devices[path] = device;
               } else {
                   device = existed;
               }
               if (device)
                   this.emit(DEVICE.CONNECT_UNACQUIRED, device.toMessageObject());
```

But that didn't make much sense :/

Especially when the error is actually `DEVICE_USED_ELSEWHERE` in my experiments, always. So this code never triggered.